### ISSUE TITLE
Dirichlet throws an unhelpful exception if ints instead of floats are passed

### ISSUE THREAD

**Issue by NatPRoach** [pymc-devs/pymc#3223]
Hello, I'm new to Pymc3, and was trying to replicate some of the examples provided, in particular, the example going over generating a mixture model of Poisson distributions provided here: https://docs.pymc.io/api/distributions/mixture.html

I found that upon running the following line
```python
w = pm.Dirichlet('w', np.array([1,1]))
```

I get the following error:

```python
log_thunk_trace: There was a problem executing an Op.
Traceback (most recent call last):
  File "./testPymc3.py", line 26, in <module>
    w = pm.Dirichlet('w', np.array([1,1]))
  File "/anaconda/lib/python2.7/site-packages/pymc3/distributions/distribution.py", line 41, in __new__
    dist = cls.dist(*args, **kwargs)
  File "/anaconda/lib/python2.7/site-packages/pymc3/distributions/distribution.py", line 52, in dist
    dist.__init__(*args, **kwargs)
  File "/anaconda/lib/python2.7/site-packages/pymc3/distributions/multivariate.py", line 417, in __init__
    (a - 1) / tt.sum(a - 1),
  File "/anaconda/lib/python2.7/site-packages/theano/tensor/var.py", line 163, in __div__
    return theano.tensor.basic.div_proxy(self, other)
  File "/anaconda/lib/python2.7/site-packages/theano/tensor/basic.py", line 3538, in div_proxy
    return int_div(x, y)
  File "/anaconda/lib/python2.7/site-packages/theano/gof/op.py", line 674, in __call__
    required = thunk()
  File "/anaconda/lib/python2.7/site-packages/theano/gof/op.py", line 862, in rval
    thunk()
  File "/anaconda/lib/python2.7/site-packages/theano/gof/cc.py", line 1735, in __call__
    reraise(exc_type, exc_value, exc_trace)
  File "<string>", line 3, in reraise
ZeroDivisionError: integer division by zero
```

This is running 
* PyMC3 Version: 3.5
* Numpy Version: 1.14.2
* Theano Version: 1.0.3
* Python Version: 2.7
* Operating system: OS X 10.11.6
* With pymc3 installed via Conda

**Comment by ckrapu** [comment]
It looks like this happens because the entries in the prior vector are integer. If you try `w = pm.Dirichlet('w', np.array([1.,1.]))`, then Numpy knows it's an array of floats and this error won't occur.

**Comment by junpenglao** [comment]
Thanks for checking @ckrapu. We should add a check/conversion inside Dirichlet.

**Comment by twiecki** [comment]
Closed by #3300.
